sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/logging/ConstructPipelineEvent.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.pipelines.logging
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.util.UUID
+
+import scala.util.control.NoStackTrace
+
+import org.apache.spark.internal.Logging
+
+case class UnresolvedFlowFailureException(name: String, cause: Throwable)
```

Comment:
let's move this to `QueryCompilationErrors` and create an error condition for it in `error-conditions.json`

Reply:
I saw that this is actually unused so I am just going to remove this for now